Robot2019/src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -10,29 +10,75 @@
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
 import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
 
+import edu.wpi.first.wpilibj.Joystick;
 import edu.wpi.first.wpilibj.command.Subsystem;
 import frc.robot.commands.TeleopDrive;
 
 public class Drivetrain extends Subsystem {
   // Put methods for controlling this subsystem
   // here. Call these from Commands.
 
-  private WPI_TalonSRX left1, right1;
-  private WPI_VictorSPX left2, left3, right2, right3;
+  private WPI_TalonSRX leftMaster, rightMaster;
+  private WPI_VictorSPX leftSlave1, leftSlave2, rightSlave1, rightSlave2;
+
+  // NOTE: Change this to a SmartDashboard Preference
+  private String drive_mode;
+
+  // NOTE: Change these names for both Arcade and Tank Drive
+  private double lvalue, rvalue;
+  private double leftK, rightK;
+
+  private Joystick leftJoy, rightJoy;
 
   public Drivetrain(WPI_TalonSRX l1, WPI_VictorSPX l2, WPI_VictorSPX l3, WPI_TalonSRX r1, WPI_VictorSPX r2,
-      WPI_VictorSPX r3) {
-    left1 = l1;
-    right1 = r1;
-    left2 = l2;
-    left3 = l3;
-    right2 = r2;
-    right3 = r3;
+      WPI_VictorSPX r3, Joystick leftJoy, Joystick rightJoy) {
+    leftMaster = l1;
+    rightMaster = r1;
+    leftSlave1 = l2;
+    leftSlave2 = l3;
+    rightSlave1 = r2;
+    rightSlave2 = r3;
+
+    this.leftJoy = leftJoy;
+    this.rightJoy = rightJoy;
+
+    drive_mode = "arcade";
+    lvalue = 0.0;
+    rvalue = 0.0;
+    leftK = 1.0;
+    rightK = 1.0;
+  }
+
+  public void TeleopDrive() {
+    lvalue = leftJoy.getY();
+    rvalue = rightJoy.getY();
+    // NOTE: The problem with this code is that it relies on the driver
+    // to not touch the right joystick if we want the robot to go forward
+    if (drive_mode.equals("arcade")) {
```

Comment:
Pot. Bug: for arcade, rvalue should be getX().

Reply:
Also, this logic does not quite match what drive mode we've been using. Try to implement the old one if possible.